scripts/lean-depot.py

```diff
@@ -0,0 +1,293 @@
+#! /usr/local/bin/python3
+import itertools
+import textwrap
+import toml
+import sys
+import os
+import argparse
+import string
+
+# * update
+# * config
+# * resolver
+# * list --broken
+# * new
+# * add
+
+if not os.path.exists('_depot'):
+    os.system('git clone --single-branch --branch snapshot https://gitlab.com/simon.hudon/lean-depot.git/ _depot')
+
+l = {}
+
+def user_command(name, parser = None):
+    if not parser:
+        parser = argparse.ArgumentParser()
+    def decorator(func):
+        parser.description = func.__doc__
+        l[name] = (func, parser)
+        return func
+    return decorator
+
+def inline_table(d):
+    res = toml.decoder.TomlDecoder()
+    res = res.get_empty_inline_table()
+    res.update(d)
+    return res
+
+def is_prefix_of(x,y):
+    return x == y[:len(x)]
+
+def snapshot_name(x):
+    (base,ext) = os.path.splitext(x)
```

Comment:
Presumably `x`, whatever that is, is expected to conform to some specific format. Is that documented anywhere?

Reply:
`snapshot-2019-10.toml` for the October snapshot